finetune/model.py

```diff
@@ -258,26 +276,42 @@ def optimizer(lr):
                     opt.apply_gradients, decay_var_list=decay_var_list
                 )
 
+                decay_var_list = [v for v in tf.global_variables() if len(v.get_shape()) > 1 or params.vector_l2]
+                opt.apply_gradients = functools.partial(opt.apply_gradients, decay_var_list=decay_var_list)
+
                 if params.dont_optimize_zero_gradients:
                     opt = dont_optimize_zeros(opt)
 
+                if params.scale_loss:
+                    opt = tf.train.experimental.MixedPrecisionLossScaleOptimizer(opt, "dynamic")
+
                 return opt
 
             summaries = (
                 tf.contrib.layers.OPTIMIZER_SUMMARIES
                 if params.summarize_grads
                 else None
             )
+            if not params.scale_loss:
```

Comment:
Figure this out. We should be able to do global norms even if we occasionally get nan gradients. Just need to remove nans before computing global norm, this step will be filtered out by the loss scaler anyway.